src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -1106,10 +1129,15 @@ rd_kafka_topic_partition_list_t *py_to_c_parts (PyObject *plist) {
 			return NULL;
 		}
 
-		rd_kafka_topic_partition_list_add(c_parts,
-						  tp->topic,
-						  tp->partition)->offset =
-			tp->offset;
+		rd_kafka_topic_partition_t *rktpar =
```

Comment:
>  Since this variable is only used in the for-loop, put it there instead.

Done

Reply:
Variable declarations need to go at the head of the block/scope to support older compilers, so move the rktpar declaration to line 1097, but keep the assignment here.

Reply:
Done